add leinigen test selectors

upgradingdave opened the pull request:
Here's another small update that I found helpful when testing the jdbc changes. This PR makes it so that `lein test` will skip tests that rely on external processes (postgres, mongo, and elasticsearch). 

In order to run the full test suite, use the following command: 

```
lein test :integration
```

danielsz commented:
This is an excellent idea. I love it. 
Let's find the appropriate keyword though. `:integration` comes from the Leiningen docs, and doesn't tell us much. Maybe `:external`or `:3rd-party`. What do you think?

upgradingdave commented:
Hmmm, yeah, good call, I think I like `:external`. Not only more descriptive, but also less to type ;-)

If you think of a better keyword, please let me know and I'd be happy to update.

danielsz commented:
I think that `:external` is still too vague. What is an external test, I would likely be asking myself.
The epiphany I had last night is `:dependency`.
```clojure
(deftest ^:dependency foo [] bar)
```
What do you think?

upgradingdave commented:
I like it, `:dependency` makes sense to me as well. I updated the PR and also added a blurb to the `Contributing` section on the README.org to describe how to run the full test suite.

danielsz commented:
So so awesome :+1:  Thanks, @upgradingdave
